gms/gossiper.cc

```diff
@@ -2560,6 +2560,13 @@ future<> gossiper::enable_features() {
 future<> persistent_feature_enabler::enable_features() {
     auto loaded_peer_features = co_await _sys_ks.load_peer_features();
     auto&& features = _g.get_supported_features(loaded_peer_features, gossiper::ignore_features_of_local_node::no);
+    std::set<std::string_view> persist;
+    for (feature& f : _feat.registered_features() | boost::adaptors::map_values) {
+        if (!f && features.contains(f.name())) {
+            persist.emplace(f.name());
+        }
+    }
```

Comment:
This logic is now repeated here and in `feature_service::enable`:
- take some list of features
- intersect it with `registered_features()` and filter out features that are already enabled

since (according to the commit message) this will be the only remaining caller of `feature_service::enable`, maybe we should remove the latter? And go over `persist` ourselves and enable the features.

Reply:
This logic deduplication can be done in a follow-up.